eis_toolkit/transformations/coda/pairwise.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pandas as pd
+from beartype import beartype
+
+from eis_toolkit.exceptions import InvalidColumnException, InvalidParameterValueException
+from eis_toolkit.utilities.checks.dataframe import check_dataframe_contains_zeros
+
+
+@beartype
+def _single_pairwise_logratio(x1: float, x2: float) -> np.float64:
+
+    return np.log(x1 / x2)
+
+
+@beartype
+def single_pairwise_logratio(x1: float, x2: float) -> np.float64:
```

Comment:
These could be typed as `Number` too to allow integers. And renamed as "numerator" and "denominator".